### PR TITLE
Recipes for R packages alakazam and prestor.

### PR DESCRIPTION
**Pull request by tfenne**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by tfenne** [comment]
Hi all.  The builds are probably going to fail on this PR.  I've been trying to get the build(s) to work, but am running into things I'm not sure how to resolve on my own.  I thought that perhaps a PR would be the easiest way to ask for help while showing what I'm trying to do.

The problem I'm having locally running `simulate-travis.py` is that r-alakazam has a ton of dependencies, which appear to get installed, and then when it attempts to load `ggplot2`.  I can't quite tell whether this is just a problem with my local setup, or a problem with the recipe.  Locally I modified the `build.sh` to push stdout and stderr to a log file that I can look at, and the error that comes out looks like this:

```
** R
** data
*** moving datasets to lazyload DB
** inst
** preparing package for lazy loading
Error: package or namespace load failed for ‘ggplot2’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/private/tmp/miniconda/conda-bld/r-alakazam_1506949613832/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/R/library/Rcpp/libs/Rcpp.so':
  dlopen(/private/tmp/miniconda/conda-bld/r-alakazam_1506949613832/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/R/library/Rcpp/libs/Rcpp.so, 6): Library not loaded: @rpath/libintl.9.dylib
  Referenced from: /private/tmp/miniconda/conda-bld/r-alakazam_1506949613832/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/R/library/Rcpp/libs/Rcpp.so
  R
Error : package ‘ggplot2’ could not be loaded
ERROR: lazy loading failed for package ‘alakazam’
```

I'd really appreciate any pointers on how to resolve this!

**Comment by daler** [comment]
Thanks @tfenne, PRs like this are the easiest way to get help. In this case, the alakazam appears to be building fine, but presto is missing dependencies (https://travis-ci.org/bioconda/bioconda-recipes/jobs/282241378#L857).

Specifically, one of the deps is alakazam. If you add `r-alakazam` as a dependency, our build system will detect that it's a dependency included in this PR and will make sure to build it first and make it available during prestor's build.

`r-bibtex` and `r-captioner` are already in conda-forge so you can just add them as deps.

As for the ggplot2 error you mentioned, not sure. Maybe it's a local config issue? Are you using the relatively-recent `simulate-travis.py --bootstrap` to build your local env?

**Comment by tfenne** [comment]
Doh, thanks @daler!  Creating these recipes has been spread out over a couple of weeks, due to waiting for R 3.4x support, and the migration of R recipes to conda-forge.  I'm the one who created the `r-bibtex` and `r-captioner` recipes at conda-forge; it's a little embarrassing that I forgot to come back and add those as dependencies after they were published!

I am using the latest Miniconda (re-installed from scratch this morning when I was troubleshooting).  Hopefully the ggplot2 issue will not surface in the travis build and I can write it off to something odd in my environment.

**Comment by tfenne** [comment]
Hi @daler - that seems to have fixed the linux build, but the MacOS build is still failing.  I can't find it again, but I recall stumbling across some discussions somewhere of problems when combining R packages (with native libraries?) across the `defaults` and `conda-forge` channels.  I wonder if that's what's happening here?  Any ideas on how best to proceed?

**Comment by daler** [comment]
Hmm. Looks like the only dep coming in from defaults is [`r-rcpp`](https://travis-ci.org/bioconda/bioconda-recipes/jobs/282287456#L553). That implies it's not available in conda-forge. Not sure if that would be the cause of the problem though. @bioconda/osx, @bioconda/r @bgruening anyone see this kind of thing before?

**Comment by bgruening** [comment]
@tfenne r-rcpp is only available from conda-forge in a older version. https://anaconda.org/conda-forge/r-rcpp/files

If you really need to pin your recipe to the newer version that is not in conda-forge, please update the conda-forge recipe.

**Comment by tfenne** [comment]
Thanks @bgruening !  Post updating r-rcpp I've rebased this branch onto master and pushed.  It now builds correctly locally, so fingers crossed it'll pass on travis too!

**Comment by tfenne** [comment]
@daler, @bgruening: Thanks for all your help getting this working.  The builds are now passing!  Would one of you mind taking a look to see if it needs any work before merging?  Cheers!

File: recipes/r-prestor/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+echo ===========================================================================
+echo = Running build.sh in $PWD =
+echo ===========================================================================
```
Comment (daler) [inline recipes/r-prestor/build.sh]: Please remove these extra echos

File: recipes/r-alakazam/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{% set version = '0.2.7' %}
+
+{% set posix = 'm2-' if win else '' %}
+{% set native = 'm2w64-' if win else '' %}
```
Comment (daler) [inline recipes/r-alakazam/meta.yaml]: in bioconda we're not supporting windows; could you please remove these win-specific vars?

File: recipes/r-alakazam/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - r-igraph >=1.0.0
+    - r-lazyeval
+    - r-progress
+    - r-readr
+    - r-scales
+    - r-seqinr
+    - r-stringi
+    - posix                # [win]
+    - {{native}}toolchain  # [win]
+    - gcc                  # [not win]
```
Comment (daler) [inline recipes/r-alakazam/meta.yaml]: Please remove the `# [win]` lines and add `gcc` and `llvm` as shown in https://bioconda.github.io/guidelines.html#c-c

File: recipes/r-alakazam/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    - r-ggplot2 >=2.0.0
+    - r-igraph >=1.0.0
+    - r-lazyeval
+    - r-progress
+    - r-readr
+    - r-scales
+    - r-seqinr
+    - r-stringi
+    - {{native}}gcc-libs   # [win]
+    - libgcc               # [not win]
```
Comment (daler) [inline recipes/r-alakazam/meta.yaml]: same here (see https://bioconda.github.io/guidelines.html#c-c)

**Comment by daler** [comment]
@tfenne great! And thank you for the conda-forge packages and updating r-rcpp. I added a couple of minor comments here inline.

**Comment by tfenne** [comment]
Thanks @daler!  I've made all the edits you requested.

**Comment by daler** [comment]
Looks good, once tests pass again go ahead and merge (or I'll check back later and merge). Thanks!

**Comment by tfenne** [comment]
Thanks @daler!

**Pull request merged**